core/services/fluxmonitor/flux_monitor.go

```diff
@@ -194,6 +194,7 @@ func (fm *concreteFluxMonitor) AddJob(job models.JobSpec) error {
 	if job.ID == nil {
 		err := errors.New("received job with nil ID")
 		logger.Error(err)
+		fm.store.UpsertErrorFor(job.ID, "Unable to add job - job has nil ID")
```

Comment:
Perhaps these ought to be constants?

Reply:
I _do_ think this functionality would benefit from these being constants - but I kinda gave up on the idea after seeing how it looked implemented. The advantage of making these constants is that we ensure the messages aren't set dynamically. While not the end of the world, but if the error message is something like "Bad thing happened at {insert time here}" then the messages would all be unique in the DB and the node operator might see 1000 error messages about the same thing and have to dismiss them one by one :grimacing: .

The reason I ditched the idea is that we can't really enforce it in the first place because if we made a new type like
```go
type ErrorDescription string
```
with constants
```go
const fmAddJobErr ErrorDescription = "Unable to add job - job has nil ID"
```
Then someone could still just write...
```go
UpsertErrorFor(job.ID, ErrorDescription("whatever I want..."))
```

So after that I kinda just thought to heck with it, maybe we should just trust devs to use it correctly. What do you think?

Side note we could also add a validation that confirms description is equal to one of the constants. That should solve all the problems.

Reply:
Seems like you thought it through and this current version is a reasonable compromise.